lib/widgets/recent_dm_conversations.dart

```diff
@@ -0,0 +1,131 @@
+      builder: (context) => const RecentDmConversationsPage());
+  }
+
+  @override
+  State<RecentDmConversationsPage> createState() => _RecentDmConversationsPageState();
+}
+
+class _RecentDmConversationsPageState extends State<RecentDmConversationsPage> with PerAccountStoreAwareStateMixin<RecentDmConversationsPage> {
+  RecentDmConversationsView? model;
+
+  @override
+  void onNewStore() {
+    model?.removeListener(_modelChanged);
+    model = PerAccountStoreWidget.of(context).recentDmConversationsView
+      ..addListener(_modelChanged);
+  }
+
+  void _modelChanged() {
+    setState(() {
+      // The actual state lives in [model].
+      // This method was called because that just changed.
+    });
+  }
+
+  Widget _buildItem(BuildContext context, DmNarrow narrow) {
+    final allRecipientIds = narrow.allRecipientIds;
+    final store = PerAccountStoreWidget.of(context);
+    final selfUser = store.users[store.account.userId]!;
+    final recipientsSansSelf = allRecipientIds
+      .whereNot((id) => id == selfUser.userId)
+      .map((id) => store.users[id]!)
+      .toList();
+
+    final Widget title;
+    final Widget avatar;
+    switch (recipientsSansSelf.length) {
+      case 0: {
+        title = Text(selfUser.fullName);
+        avatar = Avatar(userId: selfUser.userId);
+        break;
+      }
+      case 1: {
+        final otherUser = recipientsSansSelf.single;
+        title = Text(otherUser.fullName);
+        avatar = Avatar(userId: otherUser.userId);
+        break;
+      }
+      default: {
+        // TODO(i18n): List formatting, like you can do in JavaScript:
+        //   new Intl.ListFormat('ja').format(['Chris', 'Greg', 'Alya'])
+        //   // 'Chris、Greg、Alya'
+        title = Text(recipientsSansSelf.map((r) => r.fullName).join(', '));
+        avatar = ColoredBox(color: const Color(0x33808080),
+          child: Center(
+            child: Icon(ZulipIcons.group_dm, color: Colors.black.withOpacity(0.5))));
+        break;
+      }
+    }
+
+    return InkWell(
+      onTap: () {
+        Navigator.push(context,
+          MessageListPage.buildRoute(context: context, narrow: narrow));
+      },
+      child: ConstrainedBox(constraints: const BoxConstraints(minHeight: 48),
+        child: Row(crossAxisAlignment: CrossAxisAlignment.center, children: [
+          Padding(padding: const EdgeInsets.fromLTRB(12, 8, 0, 8),
+            child: _AvatarWrapper(child: avatar)),
+          const SizedBox(width: 8),
+          Expanded(child: Padding(
+            padding: const EdgeInsets.symmetric(vertical: 4),
+            child: DefaultTextStyle(
+              style: const TextStyle(
+                fontFamily: 'Source Sans 3',
+                fontSize: 17,
+                height: (20 / 17),
+                color: Color(0xFF222222),
+              ).merge(weightVariableTextStyle(context)),
+              maxLines: 2,
+              overflow: TextOverflow.ellipsis,
+              child: title),
```

Comment:
This `title` widget is always a `Text`, and so the `DefaultTextStyle` is just applying to that, right?

In that case it seems more straightforward to just have the `Text` widget here. The switch above can compute a `final String title` instead of a Widget.